### ISSUE TITLE
x86 Windows7 3.x buildbot has the issue #26624 bug

### ISSUE THREAD

**Issue by vstinner** [python/cpython#74359]
BPO | [30173](https://bugs.python.org/issue30173)
--- | :---
Nosy | @pfmoore, @db3l, @vstinner, @tjguk, @zware, @zooba

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-05-02.13:56:35.549>
created_at = <Date 2017-04-26.14:50:09.737>
labels = ['3.7', 'tests', 'OS-windows']
title = 'x86 Windows7 3.x buildbot has the issue python/cpython#70811 bug'
updated_at = <Date 2017-05-02.13:56:35.548>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-02.13:56:35.548>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-05-02.13:56:35.549>
closer = 'vstinner'
components = ['Tests', 'Windows']
creation = <Date 2017-04-26.14:50:09.737>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 30173
keywords = ['buildbot']
message_count = 6.0
messages = ['292356', '292490', '292497', '292498', '292499', '292758']
nosy_count = 6.0
nosy_names = ['paul.moore', 'db3l', 'vstinner', 'tim.golden', 'zach.ware', 'steve.dower']
pr_nums = []
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30173'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
It looks like the x86 Windows7 3.x buildbot has the issue bpo-26624 bug: test__locale hangs.

This buildbot slave is managed by David Bolen.

@David: please see:
* https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk
* https://bugs.python.org/issue26624#msg270695


http://buildbot.python.org/all/builders/x86%20Windows7%203.x/builds/549/steps/compile/logs/stdio

ValidateUcrtbase:
  setlocal
  set PYTHONPATH=D:\\cygwin\\home\\db3l\\buildarea\\3.x.bolen-windows7\\build\\Lib
  "D:\\cygwin\\home\\db3l\\buildarea\\3.x.bolen-windows7\\build\\PCBuild\\win32\\python_d.exe" "D:\\cygwin\\home\\db3l\\buildarea\\3.x.bolen-windows7\\build\\PC\\validate_ucrtbase.py" ucrtbased
  C:\\Windows\\system32\\ucrtbased.dll is version 10.0.10240.16384
  WARN: ucrtbased contains known issues. Please update the Windows 10 SDK.
  See:
    http://bugs.python.org/issue27705
    https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk


http://buildbot.python.org/all/builders/x86%20Windows7%203.x/builds/549/steps/test/logs/stdio

0:58:28 [335/404] test_base64 passed
running: test_venv (56 sec), test__locale (30 sec)
...

1:07:49 [403/404] test_asyncgen passed -- running: test__locale (562 sec)

command timed out: 3600 seconds without output running ['Tools\\\\buildbot\\\\test.bat', '-j2'], attempting to kill
running: test__locale (592 sec)
...

running: test__locale (4162 sec)
2:07:54 [404/404/1] test__locale crashed (Exit code 1)
program finished with exit code 1
elapsedTime=7680.676000

**Comment by db3l** [comment]
Ok, I believe I've got the latest ucrtbased now on the Win7 buildbot (validate_ucrtbase.py reports it as 10.0.15063.137).

It appears as if my Win8 and Win10 buildbots also have the older dll, so I'm guessing they're just fast enough or lucky enough not to run into the issue.  I'll plan on addressing them in the near future as well.

**Comment by vstinner** [comment]
@David: Oh, a build on x86 Windows7 3.5 just failed on the git step with a strange erorr:

http://buildbot.python.org/all/builders/x86%20Windows7%203.5/builds/209/steps/git/logs/err.text

twisted.spread.pb.RemoteError: unrecognized SlaveCommand 'stat'

**Comment by db3l** [comment]
Yeah, there were a few transient build errors while I was getting the buildbot back online.  My initial attempt to fix this ticket involved using the VS 2015 installer which turned out very badly, so I had to fall back to an older VM image and start over.  But the older image had an older buildbot version which didn't work with the git workflow, which hit a few pending builds before I got it fixed.

I'm not yet sure if there may be some other fallback (e.g., the link error in the recent 3.x build seems familiar but I can't recall why offhand)
